### PR TITLE
[Dashboard New IA] Add overview cards and collapsible sections to the Dashboard

### PR DESCRIPTION
**Pull request by alanwguo**
## Why are these changes needed?
Screen Shot 2022-12-15 at 2 25 55 PM

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by rkooo567 — APPROVED** [review]

**Comment by rkooo567** [comment]
Screen Shot 2022-12-15 at 7 51 39 AM

Also when there's no job, the page is a bit weird. Maybe we should have some comments?

**Comment by rkooo567** [comment]
Screen Shot 2022-12-15 at 7 53 04 AM

Another nit. Is it possible to make the legend uncut? It looks a bit ugly because it is cut in the middle

File: dashboard/client/src/pages/overview/cards/NodeCountCard.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        frameBorder="0"
+      />
+      <div className={classes.links}>
+        <Link
+          className={classNames(classes.link, classes.nodesLink)}
+          to="/new/cluster"
+        >
+          <Typography variant="h4">View all nodes →</Typography>
+        </Link>
+        <Link className={classes.link} to="">
```
Comment (rkooo567) [inline dashboard/client/src/pages/overview/cards/NodeCountCard.tsx:58]: consider removing this until we have the feature?

File: dashboard/client/src/pages/overview/cards/RecentJobsCard.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  const { jobList } = useJobList();
+  const sortedJobs = _.orderBy(jobList, ["startTime"], ["desc"]).slice(0, 6);
+
+  return (
+    <OverviewCard className={classes.root}>
+      <Typography variant="h3">Recent jobs</Typography>
+      <div className={classes.listContainer}>
+        {sortedJobs.map((job) => (
+          <RecentJobListItem
+            key={job.job_id ?? job.submission_id}
```
[the remark this answers is not in the document]
Comment (alanwguo) [reply dashboard/client/src/pages/overview/cards/RecentJobsCard.tsx:46]: yeah, great idea!

[the remark this answers is not in the document]
Comment (alanwguo) [reply dashboard/client/src/pages/overview/cards/RecentJobsCard.tsx:50]: @scottsun94 please review the design. See attached screenshot

[the remark this answers is not in the document]
Comment (scottsun94) [reply dashboard/client/src/pages/overview/cards/RecentJobsCard.tsx:50]: Looks good. How about putting them in one line instead? 
Screen Shot 2022-12-15 at 11 03 04 PM


Screen Shot 2022-12-15 at 11 06 31 PM
- We should also truncate the entrypoint value when it's getting too long (which we already do) and users could hover to see the full text.
- when I run from jupyter notebook, the entrypoint is a pretty weird, long string: (interactive_shell) /Users/huaiwei/opt/anaconda3/envs/ray/bin/python -m ipykernel_launcher -f /Users/huaiwei/Library/Jupyter/runtime/kernel-955c62d5-c0be-44aa-a4cf-789e477fd738.json

File: dashboard/client/src/pages/overview/cards/RecentJobsCard.tsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    root: {
+      display: "flex",
+      flexDirection: "row",
+      flexWrap: "nowrap",
+      alignItems: "center",
+      textDecoration: "none",
+    },
+    icon: {
+      width: 20,
+      height: 20,
```
Comment (scottsun94) [inline dashboard/client/src/pages/overview/cards/RecentJobsCard.tsx:77]: Could we use 24 instead?

**Review by scottsun94 — COMMENTED** [review]
Here are the other 5 comments.
1. RE the title of the collapsible section
- Could we make the arrow + text clickable to expand/collapse? 
- And could we change the icon to 24 *24 px?
- Could we make the title's font weight to be 500?
Screen Shot 2022-12-15 at 11 40 58 PM

2. Could we make the overview cards responsive horizontally? I.e., 24 px to the left and right of the window. The cards' width should grow/shrink according to the window/viewport's width.
Screen Shot 2022-12-15 at 11 43 30 PM

3. Could we change the border radius for the overview cards?
.MuiPaper-rounded {
    border-radius: 8px;
}

4. Could we use right arrow 24 * 24 px here instead of the character? (should be easy to swap? If not, okay to leave it out for now)

Screen Shot 2022-12-15 at 11 17 52 PM


5. RE: the graph and metrics in the first overview card.
Shall we show cluster-level metrics instead? (probably just average the usage across nodes?
I'm not sure which is more useful, so I'm open to other opinions.
Screen Shot 2022-12-15 at 11 19 24 PM

File: dashboard/client/src/pages/overview/cards/RecentJobsCard.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (alanwguo) [reply dashboard/client/src/pages/overview/cards/RecentJobsCard.tsx:50]: I prefer two lines because it gives us a little bit more space.

**Comment by alanwguo** [comment]
Comments addressed. See screenshot:

Screen Shot 2022-12-16 at 4 21 53 PM

**Pull request merged**